src/components/sign-in/signInForm.js

```diff
@@ -0,0 +1,84 @@
+import React from "react";
+
+import CustomInput from "../cunstom-input";
+import Form from "../form";
+
+const mainElements = [
+    {
+        value: "username",
+        label: "Username",
+        type: "text"
+    },
+    {
+        value: "password",
+        label: "Password",
+        type: "password"
+    },
+];
+
+
+export default class SignInForm extends React.Component {
+    constructor() {
+        super();
+        this.state = {
+            username: "",
+            password: "",
+            userId: 0
+        }
+    }
+
+    findData(username, password, body) {
+
+        for (let element of body) {
+            if (element.data.username === username)
+            {
+                if (element.data.password !== password)
+                {
+                    alert("Incorrent Password!");
+                    return false;
+                }
+
+                alert("Success!");
+                this.setState({userId: element.id});
+                return true;
+            }
+        }
+        alert("Invalid Username!");
+        return false;
+    }
+
+    validateData() {
+        fetch("/Users")
+            .then(resp => resp.json())
+            .then(body => this.findData(this.state.username, this.state.password, body))
```

Comment:
The validation of password should not be on frontend :) I can just open network tab and see all passwords :)

Reply:
This is just in case, I understand that we don't control backend